File: ecommerce/src/main/java/com/bookstore/ecommerce/app/order/query/Book.java
package com.bookstore.ecommerce.app.order.query;

import lombok.Builder;
import lombok.Data;

@Builder
@Data
public class Book {
  private String id;
  private String title;
  private String subtitle;
  private String description;
  private String thumbnailUrl;

  public static Book fromDataObject(com.bookstore.ecommerce.app.domain.data.Book book) {
    return Book.builder()
      .id(book.getId())
      .title(book.getTitle())
      .subtitle(book.getSubtitle())
      .description(book.getDescription())
      .thumbnailUrl(book.getThumbnailUrl())
      .build();
  }
}

File: order/app/domain/book.go
package domain

type BookData struct {
	Id           string
	Title        string
	Subtitle     string
	Description  string
	ThumbnailUrl string
}

func (this *BookData) Clone() BookData {
	return BookData{
		Id:           this.Id,
		Title:        this.Title,
		Subtitle:     this.Subtitle,
		Description:  this.Description,
		ThumbnailUrl: this.ThumbnailUrl,
	}
}

type Book struct {
	book BookData
}

func (Book) New(book BookData) *Book {
	return &Book{book: book}
}

func (book *Book) State() BookData {
	return book.book.Clone()
}
